test-utils/src/rpc/test_rpc.rs

```diff
@@ -25,7 +25,7 @@ impl RpcConnection for ProgramTestRpcConnection {
         payer: &Pubkey,
         signers: &[&Keypair],
         transaction_params: Option<TransactionParams>,
-    ) -> Result<Option<T>, RpcError>
+    ) -> Result<Option<(T, solana_sdk::signature::Signature)>, RpcError>
```

Comment:
A result struct would be a lot more intuitive than a tuple in my opinion.
Let's do that in a separate pr though.